services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -173,114 +174,109 @@ public RouterHttpRequestStats(
-        });
+        CollectionUtil.<TehutiMetricNameEnum, List<MeasurableStat>>mapBuilder()
+            .put(RouterTehutiMetricNameEnum.ERROR_RETRY, Collections.singletonList(new Count()))
+            .build());
 
     allowedRetryCountMetric = new MetricEntityState(
         ALLOWED_RETRY_COUNT.getMetricEntity(),
         otelRepository,
         this::registerSensorFinal,
-        new HashMap<String, List<MeasurableStat>>() {
-          {
-            put(MetricNamesInTehuti.ALLOWED_RETRY_REQUEST, Collections.singletonList(new OccurrenceRate()));
-          }
-        });
+        CollectionUtil.<TehutiMetricNameEnum, List<MeasurableStat>>mapBuilder()
+            .put(
+                RouterTehutiMetricNameEnum.ALLOWED_RETRY_REQUEST_COUNT,
+                Collections.singletonList(new OccurrenceRate()))
+            .build());
 
     disallowedRetryCountMetric = new MetricEntityState(
         DISALLOWED_RETRY_COUNT.getMetricEntity(),
         otelRepository,
         this::registerSensorFinal,
-        new HashMap<String, List<MeasurableStat>>() {
-          {
-            put(MetricNamesInTehuti.DISALLOWED_RETRY_REQUEST, Collections.singletonList(new OccurrenceRate()));
-          }
-        });
+        CollectionUtil.<TehutiMetricNameEnum, List<MeasurableStat>>mapBuilder()
+            .put(
+                RouterTehutiMetricNameEnum.DISALLOWED_RETRY_REQUEST_COUNT,
+                Collections.singletonList(new OccurrenceRate()))
+            .build());
 
     retryDelayMetric = new MetricEntityState(
         RETRY_DELAY.getMetricEntity(),
         otelRepository,
         this::registerSensorFinal,
-        new HashMap<String, List<MeasurableStat>>() {
-          {
-            put(MetricNamesInTehuti.RETRY_DELAY, Arrays.asList(new Avg(), new Max()));
-          }
-        });
+        CollectionUtil.<TehutiMetricNameEnum, List<MeasurableStat>>mapBuilder()
+            .put(RouterTehutiMetricNameEnum.RETRY_DELAY, Arrays.asList(new Avg(), new Max()))
+            .build());
 
     abortedRetryCountMetric = new MetricEntityState(
         ABORTED_RETRY_COUNT.getMetricEntity(),
         otelRepository,
         this::registerSensorFinal,
-        new HashMap<String, List<MeasurableStat>>() {
-          {
-            put(MetricNamesInTehuti.DELAY_CONSTRAINT_ABORTED_RETRY_REQUEST, Collections.singletonList(new Count()));
-            put(MetricNamesInTehuti.SLOW_ROUTE_ABORTED_RETRY_REQUEST, Collections.singletonList(new Count()));
-            put(MetricNamesInTehuti.RETRY_ROUTE_LIMIT_ABORTED_RETRY_REQUEST, Collections.singletonList(new Count()));
-            put(MetricNamesInTehuti.NO_AVAILABLE_REPLICA_ABORTED_RETRY_REQUEST, Collections.singletonList(new Count()));
-          }
-        });
+        CollectionUtil.<TehutiMetricNameEnum, List<MeasurableStat>>mapBuilder()
+            .put(
+                RouterTehutiMetricNameEnum.DELAY_CONSTRAINT_ABORTED_RETRY_REQUEST,
+                Collections.singletonList(new Count()))
```

Comment:
Nit: you might want to do a static import of the `singletonList` function to make the code shorter and fit in one line.

Reply:
Modified this. Also looked into importing `MetricNamesInTehuti` statically, but it had some overlaps with otel names leading to ambiguity, so left it for now.